resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/local/LocalFile.java

```diff
@@ -337,21 +323,25 @@ private IStatus createErrorStatus(File target, String msg, Exception e) {
 
 	@Override
 	public boolean isParentOf(IFileStore other) {
-		if (!(other instanceof LocalFile))
+		if (!(other instanceof LocalFile otherFile)) {
 			return false;
+		}
 		String thisPath = filePath;
-		String thatPath = ((LocalFile) other).filePath;
+		String thatPath = otherFile.filePath;
 		int thisLength = thisPath.length();
 		int thatLength = thatPath.length();
 		//if equal then not a parent
-		if (thisLength >= thatLength)
+		if (thisLength >= thatLength) {
 			return false;
+		}
 		if (getFileSystem().isCaseSensitive()) {
-			if (thatPath.indexOf(thisPath) != 0)
+			if (!thatPath.startsWith(thisPath)) {
 				return false;
+			}
 		} else {
-			if (thatPath.toLowerCase().indexOf(thisPath.toLowerCase()) != 0)
+			if (!thatPath.toLowerCase(Locale.ENGLISH).startsWith(thisPath.toLowerCase(Locale.ENGLISH))) {
```

Comment:
Because it is in general recommended to not rely on the default local as per JavaDoc of that method or https://stackoverflow.com/a/11063161/14542697.

However in this case this could be reconsidered since it should mimic the File-System behavior it should of course act like that, but I cannot tell what the the behavior of Windows is in Turkish.

Reply:
The javadoc [1] tells to use Locale.ROOT. I don't know how turkish OSs are implemented either. But you should not silently do functional changes without a reason nor without testcase. Please fix or revert.
[1] https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/String.html#toLowerCase()

Reply:
> The javadoc [1] tells to use Locale.ROOT.

I'm not sure if that's the right choice in this case. On first sight it actually seams reasonable to compare the names case-insensitive based on the current local, as long as the OS works the same.
But since nobody can tell this at the moment I revert it to the previous state (but I'm also not sure if it was coded this way in full awareness of the situation):
- https://github.com/eclipse-platform/eclipse.platform/pull/1501